enterprise/app/org/join_org.tsx

```diff
@@ -45,61 +47,67 @@ export default class JoinOrgComponent extends React.Component<JoinOrgComponentPr
     router.navigateHome();
   }
 
-  private async onYesClicked() {
+  private async onYesClicked(org: grp.GetGroupResponse) {
     this.setState({ status: "JOINING_GROUP" });
     try {
-      console.debug("Joining group", this.state.org.id);
-      await rpcService.service.joinGroup(Object.assign(new grp.JoinGroupRequest(), { id: this.state.org.id }));
+      console.debug("Joining group", org.id);
+      await rpcService.service.joinGroup(Object.assign(new grp.JoinGroupRequest(), { id: org.id }));
     } catch (e) {
       const error = BuildBuddyError.parse(e);
       if (error.code === "AlreadyExists") {
-        this.setState({ status: "ALREADY_EXISTS", error });
+        this.setState({ status: "ALREADY_EXISTS", error: error.description });
         return;
       } else {
         throw e;
       }
     }
 
-    if (!this.isInOrgDomain()) {
+    if (!this.isInOrgDomain(org)) {
       this.setState({ status: "REQUEST_SUBMITTED" });
       return;
     }
 
-    await authService.setSelectedGroupId(this.state.org.id);
+    await authService.setSelectedGroupId(org.id);
     router.navigateHome();
   }
 
-  private async onViewBuildsClicked() {
-    await authService.setSelectedGroupId(this.state.org.id);
+  private async onViewBuildsClicked(org: grp.GetGroupResponse) {
+    await authService.setSelectedGroupId(org.id);
     router.navigateHome();
   }
 
-  private isInOrgDomain() {
-    return this.state.org.ownedDomain === getUserEmailDomain(this.props.user);
+  private isInOrgDomain(org: grp.GetGroupResponse) {
+    return org.ownedDomain === getUserEmailDomain(this.props.user);
   }
 
   render() {
-    const { status, org } = this.state;
+    const status = this.state.status;
+    const org = this.state.org;
 
-    switch (status) {
-      case "INITIAL_LOAD":
+    if (!org) {
```

Comment:
Fair enough -- moved back to the flat `switch {}` structure and am throwing an error now in the cases where we expect the org to be defined, so that the typechecker stops complaining.